cmd/archeio/main.go

```diff
@@ -60,28 +103,50 @@ func main() {
 	}()
 	klog.Infof("Server started")
 
+	klog.InfoS("metrics", "port", metricsPort)
+	metricsServer := &http.Server{
+		Addr:        ":" + metricsPort,
+		Handler:     promhttp.Handler(),
+		ReadTimeout: 10 * time.Second,
+	}
+
+	go func() {
+		if err := metricsServer.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			klog.Fatal(err)
+		}
+	}()
+	klog.Infof("Metrics server started")
+
 	// Graceful shutdown
 	<-done
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 	defer cancel()
 
+	klog.Infof("Shutting down server")
 	if err := server.Shutdown(ctx); err != nil {
 		klog.Fatalf("Server didn't exit gracefully %v", err)
 	}
+	if err := metricsServer.Shutdown(ctx); err != nil {
+		klog.Fatalf("Metrics server didn't exit gracefully %v", err)
+	}
 }
 
 func handler(w http.ResponseWriter, r *http.Request) {
 	// right now we just need to serve a redirect, but all
 	// valid requests should be at /v2/ or /v1/, so we leave this check
 	// in the future we will selectively redirect clients to different copies
+	method := r.Method
 	path := r.URL.Path
 	switch {
 	case strings.HasPrefix(path, "/v2/"):
+		serverMetrics.Requests.With(prometheus.Labels{"code": fmt.Sprintf("%v", http.StatusPermanentRedirect), "method": method, "version": "v2"}).Inc()
 		doV2(w, r)
 	case strings.HasPrefix(path, "/v1/"):
+		serverMetrics.Requests.With(prometheus.Labels{"code": fmt.Sprintf("%v", http.StatusPermanentRedirect), "method": method, "version": "v2"}).Inc()
```

Comment:
Version=v1 in the end of the string?

Reply:
Thank you @rikatz, I missed this. Fixed in 3a50416